### PR TITLE
[GPT-J P0-P1-P2] GPT-J Basic Experience Update

### PR DESCRIPTION
**Pull request by westfish**
### PR types

New features

### PR changes

Models

### Description

P0 Add config and unit test
P1 Add loss, return_dict, hidden and attention outputs
P2 Add inputs_embeds and past_key_values

### CODE REVIEW

File: tests/transformers/gptj/test_modeling.py [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+        self.initializer_range = initializer_range
+        self.num_labels = num_labels
+        self.num_choices = num_choices
+        self.scope = None
+        self.bos_token_id = vocab_size - 1
+        self.eos_token_id = vocab_size - 1
+        self.pad_token_id = vocab_size - 1
+
+    def get_large_model_config(self):
+        return GPTJConfig.from_pretrained("EleutherAI/gpt-j-6B")
```
Comment (sijunhe) [inline tests/transformers/gptj/test_modeling.py]: 似乎没用上？

File: paddlenlp/transformers/gptj/modeling.py [hunk 2]
```diff
@@ -474,27 +427,62 @@ def forward(
             attention_mask = paddle.unsqueeze(attention_mask, axis=[1, 2]).astype(paddle.get_default_dtype())
             attention_mask = (1.0 - attention_mask) * -1e4
             attention_mask.stop_gradient = True
+        # Prepare head mask if needed
+        # 1.0 in head_mask indicate we keep the head
+        # attention_probs has shape bsz x num_attention_heads x N x N
+        # head_mask has shape n_layer x batch x num_attention_heads x N x N
+        # head_mask = self.get_head_mask(head_mask, self.config.n_layer)
```
Comment (sijunhe) [inline paddlenlp/transformers/gptj/modeling.py]: 我们这里是预设传入的head_mask就是shape已经是做好了的么？

File: paddlenlp/transformers/gptj/modeling.py [hunk 3]
```diff
@@ -474,27 +427,62 @@ def forward(
...
+        all_self_attentions = () if output_attentions else None
+        all_hidden_states = () if output_hidden_states else None
+        for i, (block, layer_past) in enumerate(zip(self.h, past_key_values)):
+            if output_hidden_states:
+                all_hidden_states = all_hidden_states + (hidden_states,)
+            outputs = block(
+                hidden_states,
+                layer_past=layer_past,
+                attention_mask=attention_mask,
+                # head_mask=head_mask[i],
```
Comment (sijunhe) [inline paddlenlp/transformers/gptj/modeling.py]: rm

**Review by sijunhe — COMMENTED** [review]
写的挺好的。
主要是在正常的config升级以外还加了挺多东西，偏生成的模型我不太熟悉。要不然 @guoshengCS 也看看？

File: tests/transformers/gptj/test_modeling.py (hunk 1 above)
Comment (westfish) [reply tests/transformers/gptj/test_modeling.py]: 确实 会删除这一部分

File: paddlenlp/transformers/gptj/modeling.py (hunk 2 above)
Comment (westfish) [reply paddlenlp/transformers/gptj/modeling.py]: 我们的基类PretrainedModel里面目前还不支持get_head_mask，所以还不太能直接使用，我把这一块的代码删了吧

File: paddlenlp/transformers/gptj/modeling.py (hunk 3 above)
Comment (westfish) [reply paddlenlp/transformers/gptj/modeling.py]: ok

File: paddlenlp/transformers/gptj/modeling.py (hunk 2 above)
Comment (sijunhe) [reply paddlenlp/transformers/gptj/modeling.py]: 是的，我们其他模型都没有支持head_mask, 可以考虑把所有带head mask的删掉

Comment (westfish) [reply paddlenlp/transformers/gptj/modeling.py]: 好的 最新的commit已经把test和modeling的head_mask全部删除

**Review by sijunhe — APPROVED** [review]
lgtm
